src/libexpr/eval.cc

```diff
@@ -173,7 +173,11 @@ void Value::print(const SymbolTable & symbols, std::ostream & str,
     case tFloat:
         str << fpoint;
         break;
+    case tBlackhole:
+        str << "«potential infinite recursion»";
```

Comment:
Maybe "potential" can be dropped, since in eval output this denotes an actual infinite recursion.